.pnpmfile.cjs

```diff
@@ -0,0 +1,69 @@
+// .pnpmfile.cjs is a pnpm hook file that allows modification of package.json content
+// (in memory) of the packages that are being installed.
+// https://pnpm.io/pnpmfile
+
+// This implementation is based on https://gist.github.com/dvins/33b8fb52480149d37cdeb98890244c5b.
+// Changes:
+//  - Added support for complete dependency removal (specify null for newVersion)
+//  - Specify remapPeerDependencies for local needs
+
+//@ts-check
+
+// https://pnpm.io/pnpmfile
+// https://github.com/pnpm/pnpm/issues/4214
+// https://github.com/pnpm/pnpm/issues/5391
+
+const rootPkg = require("./package.json");
+
+console.log(`Checking for package peerDependency overrides`);
+
+const remapPeerDependencies = [
+	// @fluidframework/azure-client 1.x declares a peerDependency on fluid-framework but does not require it.
+	// It should have been an optional peer dependency. We just remove it.
+	{
+		package: "@fluidframework/azure-client",
+		packageVersion: "1.",
+		peerDependency: "fluid-framework",
+		newVersion: null,
+	},
+];
+
+function overridesPeerDependencies(pkg) {
+	if (pkg.peerDependencies) {
+		remapPeerDependencies.map((dep) => {
```

Comment:
Nit: I'm partial to using `for...of` (or `.foreach()`, the linter shouldn't complain about it here) instead of `map` when it's just for iteration purposes and the mapping function is not returning something (so, not really _mapping_ the array items to something else). Not a blocker.